src/commands/Scheduled.ts

```diff
@@ -0,0 +1,116 @@
+import { SlashCommandBuilder } from '@discordjs/builders';
+import { CommandInteraction } from 'discord.js';
+import Command from '../Command';
+import { BotClient } from '../types';
+import { DateTime } from 'luxon';
+import schedule from 'node-schedule';
+import { v4 as newUUID } from 'uuid';
+import Logger from '../utils/Logger';
+
+/**
+ * This command will send a scheduled message after waiting
+ * for a user specified time.
+ * 
+ * 
+ * The command will run a cronjob to send the message. Currently
+ * there is a 1 day limit on scheduled messages but that will likely
+ * change soon.
+ */
+export default class Scheduled extends Command {
+  constructor(client: BotClient) {
+    const definition = new SlashCommandBuilder()
+      .setName('schedulesend')
+      .addStringOption((option) => option.setName('message')
+        .setDescription('message to be sent')
+        .setRequired(true))
+      .addStringOption((option) => option.setName('wait')
+        .setDescription('Time to wait before sending message. Use hh:mm format!')
+        .setRequired(true))
+      .setDescription('sends a scheduled message to the channel');
+
+ 
+    super(client, {
+      name: 'schedulesend',
+      boardRequired: true,
+      enabled: true,
+      description: 'sends a scheduled message',
+      category: 'Utility',
+      usage: client.settings.prefix.concat('schedulesend'),
+      requiredPermissions: ['SEND_MESSAGES'],
+    }, definition);
+  }
+  
+  public async run(interaction: CommandInteraction): Promise<void> {
+
+    /**
+     * User input values
+     */
+    const message = interaction.options.getString('message', true);
+    const member = interaction.member;
+    const wait = interaction.options.getString('wait', true);
+
+    /**
+     * Arrays to help check whether time values are valid
+     * and to configure Date object
+     */
+    let timeArray = wait.split(':'); 
+
+    //Checks if the timestring is valid
+    //If there exists a timestring
+    if (wait) {
```

Comment:
There's a code style called guard clauses that I really like using. Basically when we have nested-if conditions for validation and error handling, we can often times invert the conditions to maintain flatter code.

Instead of writing 
```
if (wait) {
    // do something that depends on wait
}
```

we can write 
```
if (!wait) {
  // throw an error or warning message
  return; // immediately terminate
}
// do something that depends on wait being defined
```

Since we expect wait to always be defined (can't pass in an empty string), this is the most common flow of logic. It's still good to provide validation that wait is non-empty but we can avoid additional nesting by flipping the logic.

I have no clue if this made sense tbh but this video has better examples 
https://www.youtube.com/watch?v=EumXak7TyQ0